event_orchestration.go

```diff
@@ -0,0 +1,486 @@
+	// results, the new Offset value should be currentOffset + Limit.
+	Offset uint `url:"offset,omitempty"`
+
+	// Total is the pagination parameter to request that the API return the
+	// total count of items in the response. If this field is omitted or set to
+	// false, the total number of results will not be sent back from the PagerDuty API.
+	//
+	// Setting this to true will slow down the API response times, and so it's
+	// recommended to omit it unless you've a specific reason for wanting the
+	// total count of items in the collection.
+	Total bool `url:"total,omitempty"`
+
+	SortBy string `url:"sort_by,omitempty"`
+}
+
+// ListOrchestrationsWithContext lists all the existing event orchestrations.
+func (c *Client) ListOrchestrationsWithContext(ctx context.Context, o ListOrchestrationsOptions) (*ListOrchestrationsResponse, error) {
+	v, err := query.Values(o)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.get(ctx, eoPath+"?"+v.Encode())
+	if err != nil {
+		return nil, err
+	}
+
+	var result ListOrchestrationsResponse
+	if err = c.decodeJSON(resp, &result); err != nil {
+		return nil, err
+	}
+
+	return &result, nil
+}
+
+// CreateOrchestrationWithContext creates a new event orchestration.
+func (c *Client) CreateOrchestrationWithContext(ctx context.Context, e Orchestration) (*Orchestration, error) {
+	d := map[string]Orchestration{
+		"orchestration": e,
+	}
+
+	resp, err := c.post(ctx, eoPath, d, nil)
+	return getOrchestrationFromResponse(c, resp, err)
+}
+
+// DeleteOrchestrationWithContext deletes an existing event orchestration.
+func (c *Client) DeleteOrchestrationWithContext(ctx context.Context, id string) error {
+	_, err := c.delete(ctx, eoPath+"/"+id)
+	return err
+}
+
+// GetOrchestrationOptions is the data structure used when calling the GetOrchestration API endpoint.
+type GetOrchestrationOptions struct {
+}
+
+// GetOrchestrationWithContext gets information about an event orchestration.
+func (c *Client) GetOrchestrationWithContext(ctx context.Context, id string, o *GetOrchestrationOptions) (*Orchestration, error) {
+	v, err := query.Values(o)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.get(ctx, eoPath+"/"+id+"?"+v.Encode())
+	return getOrchestrationFromResponse(c, resp, err)
+}
+
+// UpdateOrchestrationWithContext updates an existing event orchestration.
+func (c *Client) UpdateOrchestrationWithContext(ctx context.Context, id string, e Orchestration) (*Orchestration, error) {
+	d := map[string]Orchestration{
+		"orchestration": e,
+	}
+
+	resp, err := c.put(ctx, eoPath+"/"+id, d, nil)
+	return getOrchestrationFromResponse(c, resp, err)
+}
+
+func getOrchestrationFromResponse(c *Client, resp *http.Response, err error) (*Orchestration, error) {
+	if err != nil {
+		return nil, err
+	}
+
+	var target map[string]Orchestration
+	if dErr := c.decodeJSON(resp, &target); dErr != nil {
+		return nil, fmt.Errorf("Could not decode JSON response: %v", dErr)
```

Comment:
nit: could we go ahead and correct the following linter [`staticcheck`] suggestions:

```
event_orchestration.go:140:15: error strings should not be capitalized (ST1005)
event_orchestration.go:225:15: error strings should not be capitalized (ST1005)
event_orchestration.go:336:15: error strings should not be capitalized (ST1005)
event_orchestration.go:356:15: error strings should not be capitalized (ST1005)
event_orchestration.go:437:15: error strings should not be capitalized (ST1005)
```